tournamentcontrol/competition/signals/places.py

```diff
@@ -15,3 +17,41 @@ def set_ground_timezone(sender, instance, created, *args, **kwargs):
     if created and not instance.timezone:
         instance.timezone = instance.venue.timezone
         instance.save()
+
+
+@disable_for_loaddata
+def update_match_datetimes_on_venue_timezone_change(sender, instance, created, **kwargs):
+    """Update Match.datetime when Venue timezone changes."""
+    if not created:
+        # Check if timezone field changed
+        from tournamentcontrol.competition.models import Match
+        
+        try:
+            old_instance = sender.objects.get(pk=instance.pk)
+            if old_instance.timezone != instance.timezone:
+                # Update all matches played at this venue
+                matches = Match.objects.filter(play_at=instance, date__isnull=False, time__isnull=False)
```

Comment:
There is a related manager that can be used to find matches associated to both a Venue and Ground in the same manner; this would allow for a single handler.

Eg. `instance.matches.all()`